### PR TITLE
feat: add support for Mouse Lock API to MouseRangeManipulator

### PR DESCRIPTION
**Pull request by FezVrasta**
### Context

This change allows to set a `usePointerLock` flag to the `MouseRangeManipulator` so that the mouse is hidden while the user is interacting with the view.

This is especially useful because users may move their mouse outside the view while dragging their cursor to quickly interact with the view.

The PR is based on https://github.com/Kitware/vtk-js/pull/1272

### Results

Before, the user would keep seeing their cursor as they dragged it around, if the mouse went outside the view it could cause different issues since it could interfere with other UI elements.

After, the cursor is hidden while the user holds their button down, preventing the above issues.

### Changes

- [ ] Documentation and TypeScript definitions were updated to match those changes

I couldn't find TS definitions for this module.

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

- [ ] This change adds or fixes unit tests 
- [ ] Tested environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**: 

### Funding
This contribution is funded by [Sirona Medical](https://sironamedical.com).

### CODE REVIEW

**Comment by floryst** [comment]
LGTM. @jourdain we may need to look at how we can support this kind of interaction with the interactor, or if we need to at all.

File: Sources/Interaction/Manipulators/MouseRangeManipulator/index.js [hunk 1]
```diff
@@ -164,6 +164,63 @@ function vtkMouseRangeManipulator(publicAPI, model) {
...
+      publicAPI.startPointerLockInteraction(interactor);
+    }
+  };
+
+  publicAPI.onButtonUp = () => {
+    publicAPI.exitPointerLock();
+  };
+
+  publicAPI.bindEvents = (container) => {
+    document.addEventListener(
```
Comment (floryst) [inline Sources/Interaction/Manipulators/MouseRangeManipulator/index.js:179]: This is already handled in the [RenderWindowInteractor bindEvents](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/RenderWindowInteractor/index.js#L243), and so it shouldn't be necessary here.

File: Sources/Interaction/Manipulators/MouseRangeManipulator/index.js [hunk 2]
```diff
@@ -164,6 +164,63 @@ function vtkMouseRangeManipulator(publicAPI, model) {
...
+      subscription?.unsubscribe();
+    };
+    subscription = interactor?.onEndPointerLock(endInteraction);
+  };
+
+  publicAPI.onPointerLockMove = (e) => {
+    model.previousPosition.x += e.movementX;
+    model.previousPosition.y += e.movementY;
+
+    publicAPI.onMouseMove(null, null, model.previousPosition);
```
Comment (floryst) [inline Sources/Interaction/Manipulators/MouseRangeManipulator/index.js:211]: In thinking about this, I'm thinking that we can instead cache the interactor and renderer in onButtonDown, and used those to pass into onMouseMove here.

**Review by floryst — COMMENTED** [review]
I did a second pass, and have two comments/changes.

File: Sources/Interaction/Manipulators/MouseRangeManipulator/index.js (hunk 2 above)
Comment (floryst) [reply Sources/Interaction/Manipulators/MouseRangeManipulator/index.js:211]: And then clear said cache in onButtonRelease/Up (I forget what it's called).

Comment (FezVrasta) [reply Sources/Interaction/Manipulators/MouseRangeManipulator/index.js:211]: Why do we care if they are not used by onMouseMove?

Comment (floryst) [reply Sources/Interaction/Manipulators/MouseRangeManipulator/index.js:211]: In thinking generally, the caching pattern would be useful. But you are right, they aren't used in MouseRangeManipulator.onMouseMove, so keeping the nulls is fine.

File: Sources/Interaction/Manipulators/MouseRangeManipulator/index.js (hunk 1 above)
Comment (FezVrasta) [reply Sources/Interaction/Manipulators/MouseRangeManipulator/index.js]: It should be good now

File: Sources/Interaction/Manipulators/MouseRangeManipulator/index.js [hunk 3]
```diff
@@ -166,11 +166,64 @@ function vtkMouseRangeManipulator(publicAPI, model) {
...
   publicAPI.onMouseMove = (interactor, renderer, position) => {
     if (!model.verticalListener && !model.horizontalListener) {
       return;
     }
+
+    // We only want to initialize the pointer lock listener
+    // after the user starts moving their mouse, this way
+    // we don't interfere with other events such as doubleClick,
+    // for this reason we don't call this from `onButtonDown`
+    if (model.usePointerLock && !interactor.isPointerLocked()) {
```
[the remark this answers is not in the document]
Comment (FezVrasta) [reply Sources/Interaction/Manipulators/MouseRangeManipulator/index.js:222]: done!

**Comment by floryst** [comment]
Thanks! Looks good!

**Pull request merged**